File: src/CarRentService.java
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;
public class CarRentService {
    private List<Car> cars;
    private List<Rent> rents;

    public CarRentService(List<Car> cars, List<Rent> rents) {
        this.cars = cars;
        this.rents = rents;
    }
    public void addCar(Car car){
        cars.add(car);
    }
    public List<Car> getAvailableCars(){
        List<Car> availableCars=new ArrayList<>();
        for (Car car:cars){
            if (car.isAvailable()){
                availableCars.add(car);
            }
        }
        return availableCars;
    }

    public List<Car> getAllCars() {
        return cars;
    }
    public List<Rent> getRentalHistory() {
        return rents;
    }
    public boolean rentCar(String licensePlate, Customer customer,int duration){
        for (Car car:cars){
            if(car.getLicensePlate().equals(licensePlate)&& car.isAvailable()){
                car.setAvailability(false);
                rents.add(new Rent(car,customer, LocalDate.now(),duration));
                return  true;
            }
        }
        return false;
    }
    public void returnCar(String licensePlate){
        rents.removeIf(rental->{
            if(rental.getCar().getLicensePlate().equals(licensePlate)){
                rental.getCar().setAvailability(true);
                return true;
            }
            return false;
        });
    }
}

File: CarRentService.cs
using System;
using System.Collections.Generic;

public class CarRentService
{
    private List<Car> cars = new List<Car>();
    private List<Rent> rentalHistory = new List<Rent>();

    public void AddCar(Car car)
    {
        cars.Add(car);
    }

    public List<Car> GetAvailableCars()
    {
        return cars.FindAll(car => car.IsAvailable);
    }

    public List<Rent> GetRentalHistory()
    {
        return rentalHistory;
    }

    public bool RentCar(string licensePlate, Customer customer, int duration)
    {
        Car car = cars.Find(c => c.LicensePlate == licensePlate);
        if (car != null && car.IsAvailable)
        {
            Rent rent = new Rent(car, customer, DateTime.Now, duration);
            rentalHistory.Add(rent);
            car.IsAvailable=false;
            return true;
        }
        return false;
    }

    public void ReturnCar(string licensePlate)
    {
        Car car = cars.Find(c => c.LicensePlate == licensePlate);
        if (car != null)
        {
            car.IsAvailable=true;
        }
    }

    public List<Car> GetAllCars()
    {
        return cars;
    }
}
